### PR TITLE
Update Maven to 3.8.5 and maven-invoker to 3.1.0

### LINKED ISSUE #24643: Cannot create examples with Maven 3.8.5
### Describe the bug

Unable to create projects using Maven 3.8.5

### Expected behavior

The example project has been created successfully

### Actual behavior

```[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.511 s
[INFO] Finished at: 2022-03-30T16:59:48+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.5.Final:create (default-cli) on project standalone-pom: Execution default-cli of goal io.quarkus.platform:quarkus-maven-plugin:2.7.5.Final:create failed: An API incompatibility was encountered while executing io.quarkus.platform:quarkus-maven-plugin:2.7.5.Final:create: java.lang.NoSuchMethodError: 'org.apache.maven.model.profile.activation.FileProfileActivator org.apache.maven.model.profile.activation.FileProfileActivator.setPathTranslator(org.apache.maven.model.path.PathTranslator)'
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>io.quarkus.platform:quarkus-maven-plugin:2.7.5.Final
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/

```

### How to Reproduce?

Install maven 3.8.5

```
mvn io.quarkus.platform:quarkus-maven-plugin:2.7.5.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=mvn385bugprova \
    -Dextensions="resteasy,kubernetes,jib" \
    -DnoCode
```

### Output of `uname -a` or `ver`

Darwin 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18 OpenJDK Runtime Environment Temurin-17.0.2+8 (build 17.0.2+8) OpenJDK 64-Bit Server VM Temurin-17.0.2+8 (build 17.0.2+8, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0) Maven home: /Users/lmolteni/.sdkman/candidates/maven/3.8.5 Java version: 17.0.2, vendor: Eclipse Adoptium, runtime: /Users/lmolteni/.sdkman/candidates/java/17.0.2-tem Default locale: en_GB, platform encoding: UTF-8 OS name: "mac os x", version: "12.0.1", arch: "x86_64", family: "mac"

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by famod**
Second attempt after #24285 was aborted due to https://issues.apache.org/jira/browse/MNG-7432

See also: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Maven.20Release/near/277181649

Fixes #24643

### CODE REVIEW

**Comment by famod** [comment]
There is a strange failure in my fork, let's see what CI has to say here...

**Comment by famod** [comment]
Those enforcer tests seem to be running into MNG-7432. I'll fix it tonight.

File: independent-projects/enforcer-rules/src/it/smoketest/extensions/pom.xml [hunk 1]
```diff
@@ -20,6 +20,12 @@
     <profiles>
         <profile>
             <id>BansRuntimeDependency</id>
+            <!-- activation to ork around https://issues.apache.org/jira/browse/MNG-7432 -->
```
Comment (gsmet) [inline independent-projects/enforcer-rules/src/it/smoketest/extensions/pom.xml:23]:
```suggestion
            <!-- activation to work around https://issues.apache.org/jira/browse/MNG-7432 -->
```

File: independent-projects/enforcer-rules/src/it/smoketest/integration-tests/ext1/pom.xml [hunk 2]
```diff
@@ -36,6 +36,12 @@
     <profiles>
         <profile>
             <id>SuperfluousDeploymentDep</id>
+            <!-- activation to ork around https://issues.apache.org/jira/browse/MNG-7432 -->
```
Comment (gsmet) [inline independent-projects/enforcer-rules/src/it/smoketest/integration-tests/ext1/pom.xml:39]:
```suggestion
            <!-- activation to work around https://issues.apache.org/jira/browse/MNG-7432 -->
```

**Comment by gsmet** [comment]
@famod CI looks happy. I'll let you fix the typos. Can you make sure we can still build a generated Quarkus project with Maven 3.8.4?

**Comment by famod** [comment]
@gsmet I've build the liquibase-quickstart with both 3.8.4 and 3.8.5, as well as ther sult of:
```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=mvn-test \
    -Dextensions=resteasy-reactive,jib
```
(with version set to 999-SNAPSHOT)

Btw, we could add another Maven version to the Maven test job matrix. Or we could add a new nighly job,

**Comment by aloubyansky** [comment]
Do we really want to promote 3.8.5 in the generated projects?

**Comment by famod** [comment]
> Do we really want to promote 3.8.5 in the generated projects?

I was on the fence about that. But I think we should promote the version we "depend" on.

**Comment by aloubyansky** [comment]
I thought you wanted to make sure we are only compatible with 3.8.5. I am not sure we should make it our direct dependency.

**Comment by famod** [comment]
> I thought you wanted to make sure we are only compatible with 3.8.5

Well, haven't thought about it that way around. If you think that's better I can adjust the PR.
Thing is, while [MNG-7432](https://issues.apache.org/jira/browse/MNG-7432) does affect us, other fixes in 3.8.5 might be beneficial to have... 🤔

**Comment by aloubyansky** [comment]
Unless you think other fixes outweigh the bug you hit, I wouldn't upgrade to 3.8.5. Making sure we can still build with it will be great though.

**Comment by famod** [comment]
I'll create another PR so that you guys can decide.

Btw, I don't think we can prevent [MNG-7432](https://issues.apache.org/jira/browse/MNG-7432) from happening, even wehn staying on 3.8.4 dependency-wise.

**Comment by aloubyansky** [comment]
> Btw, I don't think we can prevent [MNG-7432](https://issues.apache.org/jira/browse/MNG-7432) from happening, even when staying on 3.8.4 dependency-wise.

That's true.

**Comment by famod** [comment]
#24729 was merged instead.

**Pull request closed without merging**